### PR TITLE
Daggers throw stun gets better at range (and awful at pointblank)

### PR DESCRIPTION
**Pull request by mc-oofert**
[BALANCE]

## About the PR 
Daggers now get better with range, with a stronger disorient and stamina damage the more tiles it travels.
Ranges above 3 directly inflict a stun. Also its crap at pointblank. Still cuts people just fine though.
https://user-images.githubusercontent.com/70376633/149582328-a7a40219-6f72-4cfc-b960-354ddc0361c9.mp4
## Why's this needed? 
Some nerd rushing you with a dagger and permastunning you is never funny, especially on rev rounds.
Also, made sec drop their guns and get stunned for 5 seconds, which usually means security ends up getting stunned with their stuff, with little chance to fight back. This PR makes it better the farther it travels, and also makes it crap at pointblank throws.

## Changelog

```changelog
(u)Jimmyl
(+)Dagger stun is now based on tiles travelled, so you cant pointblank people for free instastuns.
```

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
Personally feel as if itd be better if it just made you lie down so you couldnt chainthrow it

File: code/obj/item/misc_weapons.dm [hunk 1]
```diff
@@ -480,8 +480,18 @@
 		if (ismob(usr))
 			A:lastattacker = usr
 			A:lastattackertime = world.time
-		A.changeStatus("weakened", 6 SECONDS)
-		A:force_laydown_standup()
+
+		var/R = thr.get_throw_travelled()
```
Comment (Zonespace27) [inline code/obj/item/misc_weapons.dm:484]: more specific variable name please

File: code/obj/item/misc_weapons.dm [hunk 2]
```diff
@@ -480,8 +480,18 @@
...
+
+		if(R > 3)
+			A.changeStatus("weakened", (R * 1.5) SECONDS)
+			A:force_laydown_standup()
```
Comment (Zonespace27) [inline code/obj/item/misc_weapons.dm:488]: never use :

File: code/obj/item/misc_weapons.dm [hunk 3]
```diff
@@ -480,8 +480,18 @@
 		if (ismob(usr))
 			A:lastattacker = usr
 			A:lastattackertime = world.time
-		A.changeStatus("weakened", 6 SECONDS)
-		A:force_laydown_standup()
+
+		var/R = thr.get_throw_travelled()
+
+		if(R > 3)
```
Comment (Zonespace27) [inline code/obj/item/misc_weapons.dm:486]: make this (the 3) into a variable on the knife itself

File: code/obj/item/misc_weapons.dm [hunk 4]
```diff
@@ -480,8 +480,18 @@
...
+			A.changeStatus("weakened", (R * 1.5) SECONDS)
+			A:force_laydown_standup()
+
+		else
+			var/mob/living/carbon/C = A
+			C.do_disorient(stamina_damage = (R * 10), weakened = 0, stunned = 0, disorient = (R * 5), remove_stamina_below_zero = 1)
```
Comment (Zonespace27) [inline code/obj/item/misc_weapons.dm:492]: stamina damage feels too low

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
ehh on the PR itself, review anyway

**Comment by Crackaduck** [comment]
I feel like this is too big a nerf. It's strong, but this patch give's it too little utility at close range, where many stun weapons could also be considered too strong, and leaves it with too much risk and too little reward. I feel like it would just be better to just nerf the stun time, or make it so that it can't stun people in quick succession.

**Comment by mc-oofert** [comment]
> I feel like this is too big a nerf. It's strong, but this patch give's it too little utility at close range, where many stun weapons could also be considered too strong, and leaves it with too much risk and too little reward. It would be better to just nerf the stun time, or make it so that it can't stun people in quick succession.

just use a stunbaton if you like permastunning people at melee range

File: code/obj/item/misc_weapons.dm (hunk 2 above)
Comment (mc-oofert) [reply code/obj/item/misc_weapons.dm:489]: doesnt work without it

**Pull request closed without merging**